forge-gui/src/main/java/forge/player/HumanCostDecision.java

```diff
@@ -1002,30 +1010,73 @@ protected boolean onCardSelected(final Card c, final List<Card> otherCardsToSele
                 return false;
             }
 
-            CounterType cType = this.counterType;
-            if (cType == null) {
-                Map<CounterType, Integer> cmap = counterTable.filterToRemove(c);
+            if (oneSource) {
+                if (removeAnyType(c, numToRemove) < numToRemove) return false;
+            } else {
+                CounterType cType = this.counterType;
+                if (cType == null) {
+                    Map<CounterType, Integer> cmap = counterTable.filterToRemove(c);
 
-                String prompt = Localizer.getInstance().getMessage("lblSelectCountersTypeToRemove");
+                    String prompt = Localizer.getInstance().getMessage("lblSelectCountersTypeToRemove");
 
-                cType = getController().chooseCounterType(Lists.newArrayList(cmap.keySet()), sa, prompt, null);
-            }
+                    cType = getController().chooseCounterType(Lists.newArrayList(cmap.keySet()), sa, prompt, null);
+                }
 
-            if (cType == null || !c.canRemoveCounters(cType)) {
-                return false;
-            }
+                if (cType == null || !c.canRemoveCounters(cType)) {
+                    return false;
+                }
 
-            if (c.getCounters(cType) <= counterTable.get(null, c, cType)) {
-                return false;
-            }
+                if (c.getCounters(cType) <= counterTable.get(null, c, cType)) {
+                    return false;
+                }
 
-            counterTable.put(null, c, cType, 1);
+                counterTable.put(null, c, cType, 1);
+            }
 
             onSelectStateChanged(c, true);
             refresh();
             return true;
         }
 
+        private int removeAnyType(final Card c, int cntToRemove) {
+            int removed = 0;
+
+            final Map<CounterType, Integer> tgtCounters = Maps.newHashMap(c.getCounters());
+            for (CounterType ct : ImmutableList.copyOf(tgtCounters.keySet())) {
+                if (!c.canRemoveCounters(ct)) {
+                    tgtCounters.remove(ct);
+                }
+            }
+
+            while (cntToRemove > 0 && !tgtCounters.isEmpty()) {
+                Map<String, Object> params = Maps.newHashMap();
+                params.put("Target", c);
+
+                String prompt = Localizer.getInstance().getMessage("lblSelectCountersTypeToRemove");
+                CounterType chosenType = getController().chooseCounterType(ImmutableList.copyOf(tgtCounters.keySet()), sa, prompt, params);
+
+                int max = Math.min(cntToRemove, tgtCounters.get(chosenType));
+                // remove selection so player can't cheat additional trigger by choosing the same type multiple times
```

Comment:
I guess you added this instead because the current way would be more annoying to have you click X times?
anyway cheating isn't a concern here (unlike the copied part) because it just fills the table